infrastructure/network/addressmanager/addressrandomize.go

```diff
@@ -9,38 +10,53 @@ import (
 
 // AddressRandomize implement addressRandomizer interface
 type AddressRandomize struct {
-	random *rand.Rand
+	random         *rand.Rand
+	maxFailedCount uint64
 }
 
 // NewAddressRandomize returns a new RandomizeAddress.
-func NewAddressRandomize() *AddressRandomize {
+func NewAddressRandomize(maxFailedCount uint64) *AddressRandomize {
 	return &AddressRandomize{
-		random: rand.New(rand.NewSource(time.Now().UnixNano())),
+		random:         rand.New(rand.NewSource(time.Now().UnixNano())),
+		maxFailedCount: maxFailedCount,
 	}
 }
 
-// RandomAddress returns a random address from input list
-func (amc *AddressRandomize) RandomAddress(addresses []*appmessage.NetAddress) *appmessage.NetAddress {
-	if len(addresses) > 0 {
-		randomIndex := rand.Intn(len(addresses))
-		return addresses[randomIndex]
+// Help function which returns a random index in the range [0, len(weights)-1] with probability weighted by `weights`
```

Comment:
In go it's usually the case that every comment on function starts with the function name ("weightedRand is a help function which..."_